addons/OPT/macros.hpp

```diff
@@ -43,3 +43,83 @@
+	#define ASSERT_IS_NOT_NIL(var) if (isNil {var}) then {ASSERTION_FAILED("Variable was nil but shouldn't have been");};
+	#define ASSERT_IS_POSITION(var) ASSERT_IS_NOT_NIL(var); if !(IS_POSITION(var)) then {private _errMsg = format ["Expected Position but got %1", var]; ASSERTION_FAILED(_errMsg);};
+	#define ASSERT_IS_POSITION2D(var) ASSERT_IS_NOT_NIL(var); if !(IS_POSITION2D(var)) then {private _errMsg = format ["Expected Position2D but got %1", var]; ASSERTION_FAILED(_errMsg);};
+	#define ASSERT_IS_POSITION3D(var) ASSERT_IS_NOT_NIL(var); if !(IS_POSITION3D(var)) then {private _errMsg = format ["Expected Position3D but got %1", var]; ASSERTION_FAILED(_errMsg);};
+	#define ASSERT_IS_INTEGER(var) ASSERT_IS_NOT_NIL(var); if !(IS_SCALAR(var)) then {private _errMsg = format["Expected integer but got %1", typeName var]; ASSERTION_FAILED(_errMsg);}\
+		else { if !(floor var isEqualTo var ) then { private _errMsg = format["Expected integer but got floating point number (%1)", var]; ASSERTION_FAILED(_errMsg);};};
+	#define ASSERT_TRUE(var) ASSERT_IS_NOT_NIL(var); if !(var isEqualTo true) then {private _errMsg = format["Expected True but got %1", var]; ASSERTION_FAILED(_errMsg);};
+	#define ASSERT_FALSE(var) ASSERT_IS_NOT_NIL(var); if !(var isEqualTo false) then {private _errMsg = format["Expected False but got %1", var]; ASSERTION_FAILED(_errMsg);};
+	#define ASSERT_ARRAY_LENGTH(var,length) ASSERT_IS_NOT_NIL(var); ASSERT_IS_ARRAY(var); if !(IS_ARRAY_LENGTH(var,length)) then {private _errMsg = format ["Expected array of length but got %1", var];};
+	#define ASSERT_ALIVE(var) ASSERT_IS_NOT_NIL(var); ASSERT_IS_OBJECT(var); if !(alive var) then {ASSERTION_FAILED("Expected object to be alive but it wasn't")};
+	#define ASSERT_DEAD(var) ASSERT_IS_NOT_NIL(var); ASSERT_IS_OBJECT(var); if (alive var) then {ASSERTION_FAILED("Expected object to be alive but it wasn't")};
+#else
+	// Disable all assertions
+	#define ASSERTION_FAILED(msg)
+
+	#define ASSERT_IS_TYPE(var, type)
+	#define ASSERT_IS_ARRAY(var)
+	#define ASSERT_IS_BOOL(var)
+	#define ASSERT_IS_CODE(var)
+	#define ASSERT_IS_CONFIG(var)
+	#define ASSERT_IS_CONTROL(var)
+	#define ASSERT_IS_DISPLAY(var)
+	#define ASSERT_IS_GROUP(var)
+	#define ASSERT_IS_OBJECT(var)
+	#define ASSERT_IS_SCALAR(var)
+	#define ASSERT_IS_NUMBER(var)
+	#define ASSERT_IS_SCRIPT(var)
+	#define ASSERT_IS_SIDE(var)
+	#define ASSERT_IS_STRING(var)
+	#define ASSERT_IS_TEXT(var)
+	#define ASSERT_IS_LOCATION(var)
+	#define ASSERT_IS_NAMESPACE(var)
+	#define ASSERT_IS_NULL(var)
+	#define ASSERT_IS_NOT_NULL(var)
+	#define ASSERT_IS_NIL(var)
+	#define ASSERT_IS_NOT_NIL(var)
+	#define ASSERT_IS_POSITION(var)
+	#define ASSERT_IS_POSITION2D(var)
+	#define ASSERT_IS_POSITION3D(var)
+	#define ASSERT_IS_INTEGER(var)
+	#define ASSERT_TRUE(var)
+	#define ASSERT_FALSE(var)
+	#define ASSERT_ARRAY_LENGTH(var,length)
+	#define ASSERT_ALIVE(var)
+	#define ASSERT_DEAD(var)
+#fi
```

Comment:
Wir sind hier nicht in Bash xD
Das muss #endif heißen

@GNCLORD-MDB hier hätte der Kommentar hin gehört - wir hatten ja gesagt, dass wir möglichst nicht einfach so in den PR kommentieren